src/common/calendar/TimePickerCalendar.js

```diff
@@ -152,95 +116,62 @@ class TimePickerCalendar extends Component {
-    // (e.g. for a week view it's always the next weeks monday);
-    const activeEnd = moment(info.view.activeEnd).subtract(1, 'day');
-
-    if (momentDate.isBefore(activeStart, 'day') || momentDate.isAfter(activeEnd, 'day')) {
-      onDateChange(activeStart.format(constants.DATE_FORMAT));
-    }
-
-    const {
-      viewType,
-    } = this.state;
-
-    if (viewType !== info.view.type) {
-      this.setState({
-        viewType: info.view.type,
-      });
-    }
-  };
-
-  onEventAllow = (dropInfo) => {
-    return this.isEventValid(dropInfo.start, dropInfo.end);
-  };
-
-  onSelectAllow = (selectInfo) => {
-    return this.isEventValid(selectInfo.start, selectInfo.end);
-  };
-
   /**
-   * isEventValid();
+   * isSelectionValid();
    * @param start {Date}
    * @param end {Date}
    * @returns {boolean}
    */
-  isEventValid = (start, end) => {
-    const { isStaff, resource } = this.props;
-    return resourceUtils.isTimeRangeReservable(resource, start, end, isStaff);
-  };
-
-  onEventDrop = (eventDropInfo) => {
-    const { event } = eventDropInfo;
+  isSelectionValid = (start, end) => {
+    const { resource, isStaff } = this.props;
 
-    this.onChange({
-      start: event.start,
-      end: event.end,
-    });
-  };
-
-  onEventResize = (eventResizeInfo) => {
-    const { event } = eventResizeInfo;
-
-    this.onChange({
-      start: event.start,
-      end: event.end,
-    });
+    const events = this.getReservedEvents();
+    return resourceUtils.isTimeRangeReservable(resource, start, end, isStaff, events);
   };
 
   onEventRender = (info) => {
+    // add cancel button for new selected event
     if (info.event.id === NEW_RESERVATION) {
       const cancelBtn = document.createElement('span');
       cancelBtn.classList.add('app-TimePickerCalendar__cancelEvent');
-      cancelBtn.addEventListener('click', () => this.onChange(null), { once: true });
+      cancelBtn.addEventListener('click', () => this.onCancel(), { once: true });
       info.el.append(cancelBtn);
     }
   }
 
   onSelect = (selectionInfo) => {
-    this.onChange({
-      start: selectionInfo.start,
-      end: selectionInfo.end,
-    });
+    const { addNotification, t, resource } = this.props;
 
-    // Hide the FullCalendar selection widget/indicator.
-    const calendarApi = this.calendarRef.current.getApi();
-    calendarApi.unselect();
+    const minPeriodEndTime = resourceUtils.getMinPeriodEndTime(
+      resource,
+      selectionInfo.start, selectionInfo.end
+    );
+    // Make sure selected time alway > min_period
+
+    const isValid = this.isSelectionValid(
+      selectionInfo.start, minPeriodEndTime
+    );
+
+    if (isValid) {
+      this.onChange({
+        start: selectionInfo.start,
+        end: minPeriodEndTime
+      });
+    } else {
+      // Display error notifications if selection is not valid
+      addNotification(selectErrorNotification(t));
```

Comment:
Do you think it would possible to just use `NotificationManager` from `react-notifications` here? It would be more in line with "let's get rid of redux stuff" that we're currently doing and it would also avoid passing down several layers of this `addNotification`-prop.